src/main/java/com/teragrep/cfe_39/consumers/kafka/queue/WritableQueue.java

```diff
@@ -0,0 +1,144 @@
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.cfe_39.consumers.kafka.queue;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.FileVisitOption;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.function.BiPredicate;
+import java.util.function.ToLongFunction;
+import java.util.stream.Stream;
+
+public class WritableQueue {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WritableQueue.class);
+
+    private final Path queueDirectory;
+    private String queueNamePrefix;
+
+    public WritableQueue(String queueDirectory) {
+        this.queueDirectory = Paths.get(queueDirectory);
+        this.queueNamePrefix = "";
+        if (!Files.isDirectory(this.queueDirectory)) {
+            throw new IllegalArgumentException("Provided path is not a " + "directory <[" + queueDirectory + "]>");
+        }
+        if (!Files.isWritable(this.queueDirectory)) {
+            throw new IllegalArgumentException("Provided path is not " + "writeable <[" + queueDirectory + "]>");
+        }
+    }
+
+    private File getNextWritableFilename() throws IOException {
+
+        try (
+                Stream<Path> files = Files.find(queueDirectory, 1, getFileMatcher(queueNamePrefix), FileVisitOption.FOLLOW_LINKS)
+        ) {
+
+            long sequenceNumber = files.mapToLong(getPathToSequenceNumberFunction()).max().orElse(0);
+
+            long nextSequenceNumber = sequenceNumber + 1;
+
+            // create next
+            return new File(
+                    queueDirectory.toAbsolutePath() + File.separator + queueNamePrefix + "." + nextSequenceNumber
+            );
+        }
+        catch (UncheckedIOException uncheckedIOException) {
+            // just retry, reader modified the directory
+            return getNextWritableFilename();
+        }
+    }
+
+    public File getNextWritableFile() throws IOException {
+        if (queueNamePrefix.isEmpty()) {
+            throw new IOException("No queueNamePrefix set");
+        }
+        else {
+            return getNextWritableFilename();
+        }
+    }
+
+    public void setQueueNamePrefix(String a) {
```

Comment:
Prefer to not use any single letter variables

Reply:
Solved in commit 1ac75a34f16a9de3aa83162015a32c617f56ec47